### PR TITLE
Port to Intel LLVM based compilers

### PR DESCRIPTION
**Pull request by dkokron**
The Intel LLVM based compilers don't support certain flags.  In particular, -no-prec-div -no-prec-sqrt and -fp-model source
See https://www.intel.com/content/www/us/en/developer/articles/guide/porting-guide-for-ifort-to-ifx.html for details.

Add logic to cmake/compiler_flags_Intel_*.cmake to handle this.

Tested on an Ubuntu-22.04 system using the 2023.0.0. release of oneAPI.

Fixes # (issue)

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [ ] `make distcheck` passes

### CODE REVIEW

**Comment by thomas-robinson** [comment]
@aerorahul can you please review this?  You are the original author of this code.

**Comment by thomas-robinson** [comment]
@dkokron I'm not sure why you skipped the template for the PR, but can you open an issue and link that issue to this PR.  Then you will have to check the boxes indicating you are aware of and followed the guidelines for contributing.

File: cmake/compiler_flags_Intel_C.cmake [hunk 1]
```diff
@@ -1,8 +1,16 @@
 # Intel C
-set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -sox -traceback")
-
-set( CMAKE_C_FLAGS_RELEASE "-qno-opt-dynamic-align -O2 -debug minimal -fp-model source")
-
-set( CMAKE_C_FLAGS_DEBUG "-O0 -g -ftrapuv")
+if(CMAKE_C_COMPILER_ID STREQUAL "Intel")
```
Comment (aerorahul) [inline cmake/compiler_flags_Intel_C.cmake:2]: This file is for providing flags for the `Intel` `C` compiler.
Please create a file `compiler_flags_IntelLLVM_C.cmake` for the `IntelLLVM` `C` family.

File: cmake/compiler_flags_Intel_Fortran.cmake [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 set(r8_flags "-real-size 64") # Fortran flags for 64BIT precision
-set(r8_flags "${r8_flags} -no-prec-div -no-prec-sqrt")
 
-# Intel Fortan
-set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fpp -fno-alias -auto -safe-cray-ptr -ftz -assume byterecl -align array64byte -nowarn -sox -traceback")
-
-set(CMAKE_Fortran_FLAGS_RELEASE "-O2 -debug minimal -fp-model source -nowarn -qoverride-limits -qno-opt-dynamic-align -qopt-prefetch=3")
-
-set(CMAKE_Fortran_FLAGS_DEBUG "-g -O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fpe0 -ftrapuv")
+if(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
```
Comment (aerorahul) [inline cmake/compiler_flags_Intel_Fortran.cmake:5]: This file is for providing flags for the `Intel` `Fortran` compiler.
Please create a file `compiler_flags_IntelLLVM_Fortran.cmake` for the `IntelLLVM` `F` family.

**Review by aerorahul — CHANGES_REQUESTED** [review]
1. `IntelLLVM` is a valid `CMAKE_<LANG>_COMPILER_ID` and instead of placing `if-elseif` logic in `compiler_flags_Intel_<LANG>.cmake`, please consider adding new files specific to IntelLLVM e.g. `compiler_flags_IntelLLVM_C.cmake` etc.
2. The top-level `CMakeLists.txt` will need an update; specifically lines 44 and 49 to allow support for `IntelLLVM` compiler family.  E.g.
```
if(NOT CMAKE_C_COMPILER_ID MATCHES "^(Intel|IntelLLVM|GNU|Clang)$")
``` 
and 
```
if(NOT CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel|IntelLLVM|GNU)$")
```
respectively
3. The file `cmake/fms_compiler_flags.cmake` should be updated to extend support to `IntelLLVM` e.g.
```
elseif(CMAKE_Fortran_COMPILER_ID MATCHES "IntelLLVM")
  include(compiler_flags_IntelLLVM_Fortran)
```
and similar for the `C` compiler.

I understand there is quite a bit of repetition between the `Intel` and `IntelLLVM` compiler flags. 
My recommendation is to keep them separate so it is absolutely clear which family is used and what those flags are.

Other than that, there are a few `WARNING` messages being placed that don't need to be and could use some attention.

Out of curiosity, I noticed `-fp-model source` is being omitted from `IntelLLVM`.  Otherwise they are identical.  Can you explain why that is the case?

If they end up being identical, we can come up with a different logic than separate files for `Intel` and `IntelLLVM`.

**Review by aerorahul — DISMISSED** [review]
Changes look good to me.  Thank you for incorporating the suggestions.

**Comment by dkokron** [comment]
I have updated the CMake files as requested.  Please review and let me know if there are any other changes needed.

**Comment by dkokron** [comment]
As to why I removed "fp-model source", the source option is no longer supported by the LLVM based compilers.  See https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/compiler-reference/compiler-options/floating-point-options/fp-model-fp.html

**Comment by aerorahul** [comment]
> As to why I removed "fp-model source", the source option is no longer supported by the LLVM based compilers. See https://www.intel.com/content/www/us/en/develop/documentation/fortran-compiler-oneapi-dev-guide-and-reference/top/compiler-reference/compiler-options/floating-point-options/fp-model-fp.html

I couldn't find any explicit reference to LLVM, but that is ok since we have separate files for the LLVM compiler ID.

**Review by aerorahul — APPROVED** [review]
Changes look good.

**Review by rem1776 — APPROVED** [review]

**Pull request merged**